pkg/sdk/validations.go

```diff
@@ -79,7 +79,7 @@ func valueSet(value interface{}) bool {
 		reflectedValue = reflectedValue.Elem()
 	}
 	switch reflectedValue.Kind() {
-	case reflect.Slice:
+	case reflect.Slice, reflect.String:
```

Comment:
hmmm, I am not sure about this change, at least add a test to TestValueSet pls

Reply:
I don't remeber particular reason of why I added it here, but here's my understanding right now. If we want to enforce that some field has to be set and we're validating it with valueSet and maybe some other validating function, we want to make sure that this field is not empty (I don't think there might be case where some value is needed, but empty string is correct value, there might be one but I cannot think about one, because then I would assume it's optional). Plus I think it's correct to assume that string is just a syntactic sugar around []byte which is a slice of bytes (which was already there - I mean reflect.Slice).

Reply:
And what about e.g. comment? Let's say, you want to set it to empty string (which is correct behaviour AFAIK). Then you cannot use valueSet for it, right?

Reply:
Hmm, yeah you're right, but today I stumbled upon similar case with slice, because sometimes you may want to set empty slice because you would like to 'unset' something with it. I think we have to either change this function to support this kind of edge cases or create new valueSet function that will be able to do it with "better" defaults or more arguments / options.